config/config.go

```diff
@@ -344,6 +344,24 @@ func (c *Configuration) EnsureFilePermissions() error {
 
 	r := regexp.MustCompile("^[a-f0-9]{8}-[a-f0-9]{4}-4[a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}$")
 
+	// add trailing slash on data directory is no trailing slash exists
+	if ! strings.HasSuffix(c.System.Data, "/") {
+		c.System.Data = c.System.Data + "/"
+	}
+
+	// create the daemon-data dir if it doesn't exist
+	p, _ := path.Split(c.System.Data)
+
+	if _, err := os.Stat(c.System.Data); err != nil {
+		// if file doesn't exist
+		if os.IsNotExist(err) {
+			//
+			if err = os.Mkdir(p, 0755); err != nil {
```

Comment:
Should probably report this error since this will cause issues anyways?

Reply:
You already report when the folder doesn't exist. This creates the folder if it doesn't exist. The err is that the folder doesn't exist. The only other thing I think it might repo is if it can't read the folder due to perms or something. I will return it then.